### PR TITLE
plan: fix bug about column prunning.

### LINKED ISSUE #2794: panic problem when using union all
mysql> select b from (select a, b from t union all select a, b from t order by a) t;
ERROR 2013 (HY000): Lost connection to MySQL server during query

### PR DESCRIPTION
**Pull request by hanfei1991**
Like query `select b from (select a,b from t union all select a, b from t order by a) t`, after column pruning, the left side of union became select a, but the right side one is still a, b, which may cause a panic.
So we add a projection to every side of union to avoid this case. Fix #2794 
@shenli @coocood @zimulala PTAL

### CODE REVIEW

**Comment by coocood** [comment]
Does this change keeps columns from prunning for non-union query?
If this is true, are there any way to only use projection for union?

**Comment by hanfei1991** [comment]
No, I won't forbid any column pruning.

**Comment by coocood** [comment]
LGTM

File: plan/plan.go [hunk 1]
```diff
@@ -48,10 +48,6 @@ const (
 	Lim = "Limit"
 	// App is the type of Apply.
 	App = "Apply"
-	// Dis is the type of Distinct.
-	Dis = "Distinct"
```
Comment (shenli) [inline plan/plan.go]: There is not distinct plan?

Comment (hanfei1991) [inline plan/plan.go]: yeah, Do you remember all distinct plan have been converted to group by ?

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@coocood PTAL

**Pull request merged**